main.tf

```diff
@@ -45,10 +45,10 @@ locals {
   enabled             = local.input.enabled
   regex_replace_chars = coalesce(local.input.regex_replace_chars, local.defaults.regex_replace_chars)
 
-  name                    = lower(replace(coalesce(local.input.name, local.sentinel), local.regex_replace_chars, local.replacement))
-  namespace               = lower(replace(coalesce(local.input.namespace, local.sentinel), local.regex_replace_chars, local.replacement))
-  environment             = lower(replace(coalesce(local.input.environment, local.sentinel), local.regex_replace_chars, local.replacement))
-  stage                   = lower(replace(coalesce(local.input.stage, local.sentinel), local.regex_replace_chars, local.replacement))
+  name                    = local.id_case == "none" ? replace(coalesce(local.input.name, local.sentinel), local.regex_replace_chars, local.replacement) : lower(replace(coalesce(local.input.name, local.sentinel), local.regex_replace_chars, local.replacement))
+  namespace               = local.id_case == "none" ? replace(coalesce(local.input.namespace, local.sentinel), local.regex_replace_chars, local.replacement) : lower(replace(coalesce(local.input.namespace, local.sentinel), local.regex_replace_chars, local.replacement))
+  environment             = local.id_case == "none" ? replace(coalesce(local.input.environment, local.sentinel), local.regex_replace_chars, local.replacement) : lower(replace(coalesce(local.input.environment, local.sentinel), local.regex_replace_chars, local.replacement))
+  stage                   = local.id_case == "none" ? replace(coalesce(local.input.stage, local.sentinel), local.regex_replace_chars, local.replacement) : lower(replace(coalesce(local.input.stage, local.sentinel), local.regex_replace_chars, local.replacement))
```

Comment:
@osterman I think we should do the case conversion here for all cases. What do you think?